### PR TITLE
✨ Constructor for Resource

### PR DESCRIPTION
**Pull request by Adirio**
Provide a constructor for `Resource` in order to better split the model from the command flag options for each plugin so that third-party can consume this API instead of creating an `Options` object.

### CODE REVIEW

**Comment by Adirio** [comment]
/hold

Waiting until #1974 is merged so that validation can also be done in the constructor.

**Comment by Adirio** [comment]
/hold cancel

#1974 was merged

**Review by camilamacedo86 — APPROVED** [review]
it shows fine for me too.

**Comment by Adirio** [comment]
With plugin phase 1.5, it has become apparent that resources should be created by the CLI and injected into the different plugins so that they all refer to the same object and they can make their own modifications. This fact greatly reduces the enhancement that this PR offers, as the creation would not use any of the provided options and later modifications can't be done using options. Closing this PR, if plugin phase 1.5 approach is not followed, it may be reopened.

**Pull request closed without merging**
